projects/fsastorefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.spec.ts

```diff
@@ -0,0 +1,273 @@
+import {
+  Component,
+  DebugElement,
+  ElementRef,
+  Input,
+  Renderer2,
+} from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { RouterTestingModule } from '@angular/router/testing';
+import { I18nTestingModule } from '@spartacus/core';
+import { NavigationNode } from '@spartacus/storefront';
+import { FSNavigationUIComponent } from './navigation-ui.component';
+
+@Component({
+  // tslint:disable
+  selector: 'cx-icon',
+  template: '',
+})
+class MockIconComponent {
+  @Input() type: string;
+}
+
+@Component({
+  selector: 'cx-generic-link',
+  template: '',
+})
+class MockGenericLinkComponent {
+  @Input() url: string | any[];
+  @Input() target: string;
+  @Input() title: string;
+}
+
+const childLength = 9;
+
+const mockNode: NavigationNode = {
+  title: 'test',
+  children: [
+    {
+      title: 'Root 1',
+      url: '/root-1',
+      children: [
+        {
+          title: 'Child 1',
+          children: [
+            {
+              title: 'Sub child 1',
+              children: [
+                {
+                  title: 'Sub sub child 1',
+                  url: '/sub-sub-child-1',
+                },
+                {
+                  title: 'Sub sub child 1',
+                  url: '/sub-sub-child-1',
+                },
+                {
+                  title: 'Sub sub child 1',
+                  url: '/sub-sub-child-1',
+                },
+                {
+                  title: 'Sub sub child 1',
+                  url: '/sub-sub-child-1',
+                },
+              ],
+            },
+          ],
+        },
+        {
+          title: 'Child 2',
+          url: '/child-2',
+        },
+      ],
+    },
+    {
+      title: 'Root 2',
+      url: '/root-2',
+    },
+  ],
+};
+
+describe('Navigation UI Component', () => {
+  let fixture: ComponentFixture<FSNavigationUIComponent>;
+  let navigationComponent: FSNavigationUIComponent;
+  let element: DebugElement;
+  let renderer: Renderer2;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [RouterTestingModule, I18nTestingModule],
+      declarations: [
+        FSNavigationUIComponent,
+        MockIconComponent,
+        MockGenericLinkComponent,
+      ],
+      providers: [],
+    }).compileComponents();
+  });
+  beforeEach(() => {
+    fixture = TestBed.createComponent(FSNavigationUIComponent);
+    renderer = fixture.componentRef.injector.get<Renderer2>(Renderer2 as any);
+    navigationComponent = fixture.debugElement.componentInstance;
+    element = fixture.debugElement;
+
+    navigationComponent.node = mockNode;
+  });
+
+  describe('calculate columns', () => {
+    beforeEach(() => {
+      navigationComponent.wrapAfter = 5;
+    });
+
+    it('should return 2 for 10', () => {
+      expect(navigationComponent.getColumnCount(10)).toEqual(2);
```

Comment:
No need to copy-paste test use cases from the navigation-ui component that is being extended. In this component we should test only methods overridden/implement by us (setclasswrapper & removeClassOnWrapper).